### ISSUE TITLE
[Research] Add Self-Supervised Contrastive Pretraining for 4-Channel EEG (+20% over raw features)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#66]
## What
Pre-train an EEG encoder using self-supervised contrastive learning (SimCLR-style) on unlabeled Muse 2 data. The encoder learns robust representations without requiring emotion labels, then is fine-tuned with labeled calibration data.

## Why
- MDPI Mathematics (2025): Self-supervised framework on a 4-channel EEG device achieves **+20.4% accuracy** over conventional features
- Addresses the #1 gap: no Muse-specific labeled emotion dataset exists — self-supervised pretraining requires NO labels
- Contrastive learning distinguishes meaningful EEG patterns from noise using augmentations
- Can leverage all collected session data (labeled or not) for pretraining

### Step 1: Implement contrastive pretraining
```python
# ml/training/self_supervised_pretrain.py
class EEGContrastivePretrainer:
    def __init__(self, encoder, projection_head):
        # SimCLR-style: two augmented views of same epoch → attract
        # Different epochs → repel
    
    def augment(self, epoch):
        # Temporal jitter (shift ±50ms)
        # Channel dropout (mask 1 of 4 channels)
        # Gaussian noise (SNR 20dB)
        # Frequency masking (zero random band)
    
    def pretrain(self, unlabeled_epochs, n_epochs=100):
        # NT-Xent loss on augmented pairs
        # Returns trained encoder
```

### Step 2: Fine-tune with labels
```python
# Freeze encoder, train linear classifier on top
# Only needs 50-100 labeled samples per class (from calibration)
```

### Files to Modify
- `ml/training/self_supervised_pretrain.py` (NEW) — contrastive pretraining pipeline
- `ml/models/emotion_classifier.py` — add self-supervised encoder as feature extractor before LightGBM

### Expected Impact
- +10-20% accuracy over raw feature engineering on 4-channel data
- Leverages unlabeled data (every session contributes to pretraining)
- Better representations transfer across users

## Complexity
MEDIUM — SimCLR is well-documented, PyTorch implementation straightforward

## References
- [Portable 4-Channel EEG with Self-Supervised Feature Learning](https://www.mdpi.com/2227-7390/13/10/1608) — MDPI Mathematics, 2025
